File: frontend/src/types/crawl.ts
export interface CrawlTask {
  id: string
  company_id: string
  company_name: string
  status: 'pending' | 'running' | 'completed' | 'failed'
  jobs_found: number
  jobs_new: number
  jobs_updated: number
  error_message: string | null
  started_at: string | null
  completed_at: string | null
  created_at: string
}

File: backend/app/schemas/crawl.py
from typing import Literal

from pydantic import BaseModel

# `agent` rewrites the crawler from scratch; `cached` runs the stored script
# and never falls back, so a caller always gets the run it asked for.
CrawlMode = Literal["agent", "cached"]


class CrawlTriggerRequest(BaseModel):
    company_id: str
    mode: CrawlMode = "agent"


class CrawlTaskOut(BaseModel):
    id: str
    company_id: str
    company_name: str
    status: str
    jobs_found: int = 0
    jobs_new: int = 0
    jobs_updated: int = 0
    error_message: str | None = None
    started_at: str | None = None
    completed_at: str | None = None
    created_at: str
